modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,156 @@
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.Before;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+public abstract class BaseTemplateUpgradeProcessTestCase {
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId(), TestPropsValues.getUserId());
+	}
+
+	protected void addDDMTemplate(String filePath) throws Exception {
+		long classPK = ParamUtil.getLong(_serviceContext, "classPK");
+
+		_ddmTemplate = _ddmTemplateService.addTemplate(
+			_group.getGroupId(), _portal.getClassNameId(DDMTemplate.class),
+			classPK, _portal.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Description"
+			).build(),
+			DDMTemplateConstants.TEMPLATE_TYPE_DISPLAY, StringPool.BLANK,
+			TemplateConstants.LANG_TYPE_FTL, read(filePath), _serviceContext);
+	}
+
+	protected void addFragmentEntry(String filePath) throws Exception {
+		FragmentCollection fragmentCollection =
+			_fragmentCollectionService.addFragmentCollection(
+				_group.getGroupId(), "Fragment Collection", StringPool.BLANK,
+				_serviceContext);
+
+		_fragmentEntry = _fragmentEntryService.addFragmentEntry(
+			_group.getGroupId(), fragmentCollection.getFragmentCollectionId(),
+			null, "FragmentEntry Name", null, read(filePath), null, false, null,
+			null, 0, FragmentConstants.TYPE_COMPONENT, null,
+			WorkflowConstants.STATUS_APPROVED, _serviceContext);
+	}
+
+	protected DDMTemplate getDDMTemplate() throws Exception {
+		return _ddmTemplateService.getTemplate(_ddmTemplate.getTemplateId());
+	}
+
+	protected FragmentEntry getFragmentEntry() throws Exception {
+		return _fragmentEntryService.fetchFragmentEntry(
+			_fragmentEntry.getFragmentEntryId());
+	}
+	
+
+	protected abstract String getUpgradeStepClassName() throws Exception;
+
+	protected String read(String filePath) throws Exception {
+		Class<?> clazz = getClass();
+
+		return StringUtil.read(
+			clazz.getClassLoader(),
+			"com/liferay/dynamic/data/mapping/dependencies/upgrade" + filePath);
+	}
+
+	protected void runTemplateUpgrade() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				getUpgradeStepClassName(), LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, getUpgradeStepClassName());
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
+		}
+	}
+
+	@Inject
+	private DDMTemplate _ddmTemplate;
+
+	@Inject
```

Comment:
@alberttgomes, no need to have this `@Inject` annotation. It's used just to inject dependencies, like Liferay Services and Local Services, for example. Remember about it. :wink: Also, take some time to understand more about this annotation.

Reply:
please @natocesarrego !! Thx